src/meta/src/barrier/schedule.rs

```diff
@@ -320,63 +323,209 @@ pub struct ScheduledBarriers {
-            num_uncheckpointed_barrier: 0,
-            checkpoint_frequency,
+            sys_barrier_interval,
+            sys_checkpoint_frequency,
+            databases,
+            timer_streams,
+        }
+    }
+
+    /// Update the system barrier interval.
+    pub(super) fn set_sys_barrier_interval(&mut self, duration: Duration) {
+        if self.sys_barrier_interval == duration {
+            return;
+        }
+        self.sys_barrier_interval = duration;
+        // Reset the `IntervalStream` for all databases that use default param.
+        for (db_id, db_state) in &mut self.databases {
+            if db_state.barrier_interval.is_none() {
+                let mut interval = tokio::time::interval(duration);
+                interval.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Delay);
+                self.timer_streams
+                    .insert(*db_id, IntervalStream::new(interval));
+            }
+        }
+    }
+
+    /// Update the system checkpoint frequency.
+    pub fn set_sys_checkpoint_frequency(&mut self, frequency: u64) {
+        if self.sys_checkpoint_frequency == frequency {
+            return;
+        }
+        self.sys_checkpoint_frequency = frequency;
+        // Reset the `num_uncheckpointed_barrier` for all databases that use default param.
+        for db_state in self.databases.values_mut() {
+            if db_state.checkpoint_frequency.is_none() {
+                db_state.num_uncheckpointed_barrier = 0;
+                db_state.force_checkpoint = false;
+            }
+        }
+    }
+
+    pub(super) fn update_database_barrier(
+        &mut self,
+        database_id: DatabaseId,
+        barrier_interval_ms: Option<u32>,
+        checkpoint_frequency: Option<u64>,
+    ) {
+        match self.databases.entry(database_id) {
+            Entry::Occupied(mut entry) => {
+                let db_state = entry.get_mut();
+                db_state.barrier_interval =
+                    barrier_interval_ms.map(|ms| Duration::from_millis(ms as u64));
+                db_state.checkpoint_frequency = checkpoint_frequency;
+                // Reset the `num_uncheckpointed_barrier` since the barrier interval or checkpoint frequency is changed.
+                db_state.num_uncheckpointed_barrier = 0;
+                db_state.force_checkpoint = false;
+            }
+            Entry::Vacant(entry) => {
+                entry.insert(DatabaseBarrierState::new(
+                    barrier_interval_ms,
+                    checkpoint_frequency,
+                ));
+            }
         }
+
+        // If the database already has a timer stream, reset it with the new interval.
+        let duration = if let Some(ms) = barrier_interval_ms {
+            Duration::from_millis(ms as u64)
+        } else {
+            self.sys_barrier_interval
+        };
+        self.timer_streams.insert(
+            database_id,
+            IntervalStream::new(tokio::time::interval(duration)),
```

Comment:
Seems that we didn't do `interval.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Delay)` like other usages.

Shall we have a unified function to create the interval stream from duration?